server/user.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+	msgraph "github.com/yaegashi/msgraph.go/beta"
+	"golang.org/x/oauth2"
+)
+
+// UserInfo defines the information we store from each user
+type UserInfo struct {
+	Email string
+
+	OAuthToken          *oauth2.Token `json:",omitempty"`
+	EncryptedOAuthToken string        `json:"encrypted_oauth_token,omitempty"`
+
+	// Mattermost userID
+	UserID string
+	// Remote userID
+	RemoteID string
+	// Remote UPN
+	UPN string
```

Comment:
It seems odd to use snake case for only one field here. I understand why, for backwards compatibility, but I would think we want to be consistent with the casing